lib/src/ruby/ruby.cc

```diff
@@ -18,14 +18,14 @@ static const char load_puppet[] =
 "  $LOAD_PATH << Puppet[:libdir]\n"
 "end\n"
 "Facter.reset\n"
+"Facter.search_external([Puppet[:pluginfactdest]])\n"
```

Comment:
Puppet's `facter` terminus calls both `Facter.search` and `Facter.search_external`. The former includes custom facts from all modules in the environment. The latter includes external facts from all modules in the environment, and the external facts in `pluginfactdest`.

This commit is only adding external fact search path for `pluginfactdest`. Do we care about omitting custom and external facts from modules? For example `facter -p` will return different facts than `puppet apply`. I think we made explicit decision to skip those, but I don't remember the details?